client/src/components/VideoCall.jsx

```diff
@@ -48,7 +50,61 @@ export function VideoCall({ roomUrl, record }) {
         console.debug("track-started", event);
       }
     });
-  };
+
+    callFrame.on("active-speaker-change", (event) => {
+      // console.log("active speaker change");
+      const speakerEvents = player.get("speakerEvents") || [];
+      stage.set("currentSpeaker", event.activeSpeaker.peerId);
+      const timestamp = Date.now();
+      let currentCumulative = 0;
+      if (speakerEvents.length !== 0) {
+        currentCumulative = speakerEvents[speakerEvents.length-1].cumulative;
+      }
+      if (lastSpeaker === callFrame.participants().local.session_id) {
+        const speakerEvent = {
+          "type": "stop",
+          "timestamp": timestamp,
+          "cumulative": currentCumulative + (timestamp - speakerEvents[speakerEvents.length-1].timestamp) / 1000,
+        };
+        player.set("speakerEvents", [...speakerEvents, speakerEvent]);
+      }
+      if (event.activeSpeaker.peerId === callFrame.participants().local.session_id) {
+        // console.log("Im speaking");
+        const speakerEvent = {
+          "type": "start",
+          "timestamp": timestamp,
+          "cumulative": currentCumulative,
+        };
+        player.set("speakerEvents", [...speakerEvents, speakerEvent]);
+       // player.append("speakerEvents", speakerEvent);
+       lastSpeaker = event.activeSpeaker.peerId;
+      } 
+      console.log(player.get("speakerEvents"));
+    }); 
+
+    callFrame.on("participant-left", (event) => {
+      if (event) {
+        const speakerEvents = player.get("speakerEvents") || [];
+        const timestamp = Date.now();
```

Comment:
these lines seem to mirror some of the code above. Would it make sense to pull them out into a separate function that we call from both callbacks?